File: DreamTeam/src/main/java/team/project/redboost/entities/ServiceP.java
package team.project.redboost.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table
public class ServiceP {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private Double price;
    private String typeservice;

    @ElementCollection
    @CollectionTable(name = "service_subservices", joinColumns = @JoinColumn(name = "service_id"))
    @Column(name = "sub_service")
    private List<String> subServices = new ArrayList<>();
}

File: Redboost_front/src/app/models/ServiceP.ts
export class ServiceP {
    id?: number;
    name: string;
    description: string;
    price: number;
    typeservice: string;
    subServices: string[];

    constructor(name: string, description: string, price: number, typeservice: string, subServices: string[] = []) {
        this.id = undefined;
        this.name = name;
        this.description = description;
        this.price = price;
        this.typeservice = typeservice;
        this.subServices = subServices;
    }
}

export class Pack {
    id?: number;
    name: string;
    description: string;
    services: ServiceP[];
    price: number;
    isHighlighted: boolean;

    constructor(name: string, description: string, services: ServiceP[] = [], price: number, isHighlighted: boolean = false) {
        this.name = name;
        this.description = description;
        this.services = services;
        this.price = price;
        this.isHighlighted = isHighlighted;
    }
}
